mirrord/agent/src/file.rs

```diff
@@ -472,6 +485,67 @@ impl FileManager {
             })
     }
 
+    pub(crate) fn mkdir(&mut self, path: &Path, mode: u32) -> RemoteResult<MakeDirResponse> {
+        trace!("FileManager::mkdir -> path {:#?} | mode {:#?}", path, mode);
+
+        let path = resolve_path(path, &self.root_path)?;
+
+        let result = std::fs::create_dir(Path::new(&path));
+
+        if let Err(err) = &result {
+            let errno = err.raw_os_error().unwrap_or(0);
+            return Ok(MakeDirResponse { result: -1, errno });
+        }
+
+        let permissions = fs::Permissions::from_mode(mode);
+        fs::set_permissions(path, permissions)?;
+
+        Ok(MakeDirResponse {
+            result: 0,
+            errno: 0,
+        })
+    }
+
+    pub(crate) fn mkdirat(
```

Comment:
Like above, you can use just one call to [`nix::sys::stat::mkdirat`](https://docs.rs/nix/latest/nix/sys/stat/fn.mkdirat.html)

Reply:
👍 Done, thanks